spdmlib/src/message/version.rs

```diff
@@ -110,9 +110,15 @@ impl SpdmCodec for SpdmVersionResponsePayload {
         let mut version_count = 0;
         let rest = r.take(version_number_entry_count as usize * 2)?;
 
-        for i in 0..version_number_entry_count {
+        'version_read_loop: for i in 0..version_number_entry_count {
             if let Some(ver) = SpdmVersionStruct::read_bytes(&rest[i as usize * 2..]) {
                 if version_count < MAX_SPDM_VERSION_COUNT {
+                    for j in 0..i {
+                        if versions[j as usize].version == ver.version {
+                            // duplicated version, just ignore it
+                            continue 'version_read_loop;
+                        }
+                    }
```

Comment:
It seems overkill to use 2nd for-loop to remove dup entry.

To remove dup, maybe we can just define a fixed index.
For example, version1.0 always in index 0, version 1.1 always in index 1, version 1.2 always in index 2.
So, it does matter how many version 1.2 are input.